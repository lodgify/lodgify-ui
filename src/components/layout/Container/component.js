import React from 'react';
import { Container as SemanticContainer } from 'semantic-ui-react';

/**
 * Container is the Lodgify UI interface for the
 * Semantic UI Container.
 * @returns {Object}
 */
export const Component = ({ textAlign, ...props }) => (
  <SemanticContainer {...props} textAlign={textAlign} />
);

Component.displayName = 'Container';
