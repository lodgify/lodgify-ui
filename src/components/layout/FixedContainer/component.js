import React from 'react';
import PropTypes from 'prop-types';

/**
 * A container for placing children in a fixed position relative to the viewport.
 */
// eslint-disable-next-line jsdoc/require-jsdoc
export const Component = ({ bottom, children, left, right, top, zIndex }) => (
  <div
    style={{
      bottom,
      display: 'inline-block',
      left,
      position: 'fixed',
      right,
      top,
      zIndex,
    }}
  >
    {children}
  </div>
);

Component.displayName = 'FixedContainer';

Component.defaultProps = {
  bottom: '',
  left: '',
  right: '',
  top: '',
  zIndex: 1,
};

Component.propTypes = {
  /** The distance between the container's bottom edge and the bottom edge of the viewport. */
  bottom: PropTypes.string,
  /** The children to be wrapped with a fixed container. */
  children: PropTypes.node.isRequired,
  /** The distance between the container's left edge and the left edge of the viewport. */
  left: PropTypes.string,
  /** The distance between the container's right edge and the right edge of the viewport. */
  right: PropTypes.string,
  /** The distance between the container's top edge and the top edge of the viewport. */
  top: PropTypes.string,
  /** The priority of the fixed container in the stacking context of the page. */
  zIndex: PropTypes.number,
};
