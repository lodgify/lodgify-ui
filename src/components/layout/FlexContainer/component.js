import React from 'react';
import PropTypes from 'prop-types';

/**
 * A container for laying out children using the
 * [flexbox method](https://developer.mozilla.org/en-US/docs/Learn/CSS/CSS_layout/Flexbox).
 * Fills the full width and height of its parent or remaining space of a flex parent.
 */
// eslint-disable-next-line jsdoc/require-jsdoc
export const Component = ({
  alignContent,
  alignItems,
  children,
  flexDirection,
  flexWrap,
  justifyContent,
}) => (
  <div
    className="flex-container"
    style={{
      display: 'flex',
      flexGrow: '1',
      height: '100%',
      alignContent,
      alignItems,
      flexDirection,
      flexWrap,
      justifyContent,
    }}
  >
    {children}
  </div>
);

Component.displayName = 'FlexContainer';

Component.defaultProps = {
  alignContent: null,
  alignItems: null,
  flexDirection: null,
  flexWrap: null,
  justifyContent: null,
};

Component.propTypes = {
  /** If there is more than one line of children, the alignment of the children as a whole relative to the cross axis of the flex container. */
  alignContent: PropTypes.oneOf([
    'flex-start',
    'flex-end',
    'center',
    'space-between',
    'space-around',
    'stretch',
  ]),
  /** The alignment of each child relative to the cross axis of the line on which that child is rendered. */
  alignItems: PropTypes.oneOf([
    'flex-start',
    'flex-end',
    'center',
    'baseline',
    'stretch',
  ]),
  /** The children to be wrapped with a flex container. */
  children: PropTypes.node.isRequired,
  /** The orientation of the main axis of the flex container. */
  flexDirection: PropTypes.oneOf([
    'row',
    'row-reverse',
    'column',
    'column-reverse',
  ]),
  /** The wrap behaviour of the children. */
  flexWrap: PropTypes.oneOf(['nowrap', 'wrap', 'wrap-reverse']),
  /** The alignment of the children as a whole relative to the main axis of the flex container. */
  justifyContent: PropTypes.oneOf([
    'flex-start',
    'flex-end',
    'center',
    'space-between',
    'space-around',
    'space-evenly',
  ]),
};
