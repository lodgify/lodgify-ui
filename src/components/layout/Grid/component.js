import React from 'react';
import PropTypes from 'prop-types';
import { Grid } from 'semantic-ui-react';

/**
 * Grid is the Lodgify UI interface for the
 * Semantic UI Grid.
 *
 * To customise how the Grid gets computed, override the @columnCount
 * definition of Semantic UI.
 */
// eslint-disable-next-line jsdoc/require-jsdoc
export const Component = ({ areColumnsCentered, isStackable, ...props }) => (
  <Grid {...props} centered={areColumnsCentered} stackable={isStackable} />
);

Component.displayName = 'Grid';

Component.defaultProps = {
  areColumnsCentered: false,
  isStackable: false,
};

Component.propTypes = {
  /** Are the columns in the grid centered. */
  areColumnsCentered: PropTypes.bool,
  /** Do the items in the grid stack on top of one another on mobile devices. */
  isStackable: PropTypes.bool,
};
