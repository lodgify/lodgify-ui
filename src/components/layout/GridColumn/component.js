import React from 'react';
import PropTypes from 'prop-types';
import { Grid } from 'semantic-ui-react';

/**
 * GridColumn is the Lodgify UI interface for the
 * Semantic UI Grid.Column.
 *
 * @returns {Object}
 */
export const Component = ({ verticalAlignContent, ...props }) => (
  <Grid.Column {...props} verticalAlign={verticalAlignContent} />
);

Component.displayName = 'GridColumn';

Component.defaultProps = {
  verticalAlignContent: 'top',
  width: 12,
};

Component.propTypes = {
  /** Vertically align the content of the column to the bottom, middle or top. */
  verticalAlignContent: PropTypes.oneOf(['bottom', 'middle', 'top']),
  /** The width of the column */
  width: PropTypes.oneOf([1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]),
};
