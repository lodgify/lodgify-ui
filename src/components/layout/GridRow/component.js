import React from 'react';
import PropTypes from 'prop-types';
import { Grid } from 'semantic-ui-react';

/**
 * GridRow is the Lodgify UI interface for the
 * Semantic UI Grid.Row.
 */
// eslint-disable-next-line jsdoc/require-jsdoc
export const Component = ({ horizontalAlignContent, ...props }) => (
  <Grid.Row {...props} textAlign={horizontalAlignContent} />
);

Component.displayName = 'GridRow';

Component.defaultProps = {
  horizontalAlignContent: 'left',
};

Component.propTypes = {
  /** Horizontally align the content of the row to the left, center or right. Text content can be justified to fit the width of the row. */
  horizontalAlignContent: PropTypes.oneOf([
    'left',
    'center',
    'right',
    'justified',
  ]),
};
