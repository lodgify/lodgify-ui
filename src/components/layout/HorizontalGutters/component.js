import React from 'react';
import { Container } from 'semantic-ui-react';

/**
 * HorizontalGutters wraps child components with responsive
 * horizontal gutters on the left and right.
 */
// eslint-disable-next-line jsdoc/require-jsdoc
export const Component = props => <Container {...props} />;

Component.displayName = 'HorizontalGutters';
