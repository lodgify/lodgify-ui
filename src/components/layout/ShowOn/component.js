import React from 'react';
import PropTypes from 'prop-types';
import getClassNames from 'classnames';

/**
 * Shows child components on specified screen sizes.
 */
// eslint-disable-next-line jsdoc/require-jsdoc
export const Component = ({
  children,
  computer,
  mobile,
  parent,
  parentProps,
  tablet,
  widescreen,
}) =>
  React.createElement(
    parent,
    {
      ...parentProps,
      className: getClassNames(parentProps.className, {
        'show-on-mobile': mobile,
        'show-on-tablet': tablet,
        'show-on-computer': computer,
        'show-on-widescreen': widescreen,
      }),
    },
    children
  );

Component.defaultProps = {
  children: null,
  computer: false,
  mobile: false,
  parent: 'div',
  parentProps: {},
  tablet: false,
  widescreen: false,
};

Component.displayName = 'ShowOn';

Component.propTypes = {
  /** The child components only to be rendered on the selected devices */
  children: PropTypes.node,
  /** Are the child components showing on computer */
  // eslint-disable-next-line react/boolean-prop-naming
  computer: PropTypes.bool,
  /** Are the child components showing on mobile */
  // eslint-disable-next-line react/boolean-prop-naming
  mobile: PropTypes.bool,
  /** The parent component that will be rendered. */
  parent: PropTypes.oneOfType([PropTypes.node, PropTypes.func]),
  /** The props to be passed to the parent component. */
  // eslint-disable-next-line react/forbid-prop-types
  parentProps: PropTypes.object,
  /** Are the child components showing on tablet */
  // eslint-disable-next-line react/boolean-prop-naming
  tablet: PropTypes.bool,
  /** Are the child components showing on widescreen */
  // eslint-disable-next-line react/boolean-prop-naming
  widescreen: PropTypes.bool,
};
