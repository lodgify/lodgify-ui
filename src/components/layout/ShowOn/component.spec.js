import React from 'react';
import { mount } from 'enzyme';
import { expectComponentToHaveDisplayName } from '@lodgify/enzyme-jest-expect-helpers';

import { Component as ShowOn } from './component';

const getShowOn = props => mount(<ShowOn {...props} />);

describe('<ShowOn />', () => {
  it('should render the right structure', () => {
    const actual = getShowOn();

    expect(actual).toMatchSnapshot();
  });

  describe('if `props.mobile` is `true`', () => {
    it('should render the right structure', () => {
      const actual = getShowOn({ mobile: true });

      expect(actual).toMatchSnapshot();
    });
  });

  describe('if `props.tablet` is `true`', () => {
    it('should render the right structure', () => {
      const actual = getShowOn({ tablet: true });

      expect(actual).toMatchSnapshot();
    });
  });

  describe('if `props.computer` is `true`', () => {
    it('should render the right structure', () => {
      const actual = getShowOn({ computer: true });

      expect(actual).toMatchSnapshot();
    });
  });

  describe('if `props.widescreen` is `true`', () => {
    it('should render the right structure', () => {
      const actual = getShowOn({ widescreen: true });

      expect(actual).toMatchSnapshot();
    });
  });

  it('should have `displayName` `ShowOn`', () => {
    expectComponentToHaveDisplayName(ShowOn, 'ShowOn');
  });
});
