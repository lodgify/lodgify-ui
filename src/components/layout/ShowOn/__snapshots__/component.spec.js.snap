// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShowOn /> if \`props.computer\` is \`true\` should render the right structure 1`] = `
<ShowOn
  computer={true}
  mobile={false}
  parent="div"
  parentProps={Object {}}
  tablet={false}
  widescreen={false}
>
  <div
    className="show-on-computer"
  />
</ShowOn>
`;

exports[`<ShowOn /> if \`props.mobile\` is \`true\` should render the right structure 1`] = `
<ShowOn
  computer={false}
  mobile={true}
  parent="div"
  parentProps={Object {}}
  tablet={false}
  widescreen={false}
>
  <div
    className="show-on-mobile"
  />
</ShowOn>
`;

exports[`<ShowOn /> if \`props.tablet\` is \`true\` should render the right structure 1`] = `
<ShowOn
  computer={false}
  mobile={false}
  parent="div"
  parentProps={Object {}}
  tablet={true}
  widescreen={false}
>
  <div
    className="show-on-tablet"
  />
</ShowOn>
`;

exports[`<ShowOn /> if \`props.widescreen\` is \`true\` should render the right structure 1`] = `
<ShowOn
  computer={false}
  mobile={false}
  parent="div"
  parentProps={Object {}}
  tablet={false}
  widescreen={true}
>
  <div
    className="show-on-widescreen"
  />
</ShowOn>
`;

exports[`<ShowOn /> should render the right structure 1`] = `
<ShowOn
  computer={false}
  mobile={false}
  parent="div"
  parentProps={Object {}}
  tablet={false}
  widescreen={false}
>
  <div
    className=""
  />
</ShowOn>
`;
